framework-plugins/lisk-framework-chain-connector-plugin/test/unit/utils.spec.ts

```diff
@@ -12,92 +12,189 @@
+					bftWeightsUpdate: [BigInt(30), BigInt(40)],
+					bftWeightsUpdateBitmap: bytesToBuffer('110'),
+				},
 			},
+		],
+		[
+			// 2 new validators and update bft weight for existing ones
 			{
-				blsKey: Buffer.from('04', 'hex'),
-				bftWeight: BigInt(40),
-				address: Buffer.from('dd', 'hex'),
+				currentValidators: [
+					{
+						blsKey: Buffer.from('02', 'hex'),
+						bftWeight: BigInt(20),
+					},
+				],
+				newValidators: [
+					{
+						blsKey: Buffer.from('02', 'hex'),
+						bftWeight: BigInt(99),
+					},
+					{
+						blsKey: Buffer.from('03', 'hex'),
+						bftWeight: BigInt(30),
+					},
+					{
+						blsKey: Buffer.from('04', 'hex'),
+						bftWeight: BigInt(40),
+					},
+				],
+				expected: {
+					blsKeysUpdate: [Buffer.from('03', 'hex'), Buffer.from('04', 'hex')],
+					bftWeightsUpdate: [BigInt(99), BigInt(30), BigInt(40)],
+					bftWeightsUpdateBitmap: bytesToBuffer('111'),
+				},
 			},
-		];
-
-		const actualValidators = getActiveValidatorsDiff(currentValidators, newValidators);
-
-		expect(actualValidators).toStrictEqual(expectedValidators);
-	});
-
-	it('should aggregate existing validators having blsKey not existing in new validators with bftWeight set to 0 and aggregating the new validator as well', () => {
-		const currentValidators: ActiveValidatorWithAddress[] = [
+		],
+		[
+			// complete new set
 			{
-				blsKey: Buffer.from('01', 'hex'),
-				bftWeight: BigInt(10),
-				address: Buffer.from('aa', 'hex'),
+				currentValidators: [
+					{
+						blsKey: Buffer.from('02', 'hex'),
+						bftWeight: BigInt(20),
+					},
+				],
+				newValidators: [
+					{
+						blsKey: Buffer.from('05', 'hex'),
+						bftWeight: BigInt(99),
+					},
+					{
+						blsKey: Buffer.from('03', 'hex'),
+						bftWeight: BigInt(30),
+					},
+					{
+						blsKey: Buffer.from('04', 'hex'),
+						bftWeight: BigInt(40),
+					},
+				],
+				expected: {
+					blsKeysUpdate: [
+						Buffer.from('03', 'hex'),
+						Buffer.from('04', 'hex'),
+						Buffer.from('05', 'hex'),
+					],
+					bftWeightsUpdate: [BigInt(0), BigInt(30), BigInt(40), BigInt(99)],
+					bftWeightsUpdateBitmap: bytesToBuffer('1111'),
+				},
 			},
+		],
+		[
+			// complete new set
 			{
-				blsKey: Buffer.from('02', 'hex'),
-				bftWeight: BigInt(20),
-				address: Buffer.from('bb', 'hex'),
+				currentValidators: [
+					{
+						blsKey: Buffer.from('02', 'hex'),
+						bftWeight: BigInt(20),
+					},
+				],
+				newValidators: new Array(100).fill(0).map((_, i) => ({
+					blsKey: Buffer.from([i + 3]),
+					bftWeight: BigInt(i + 10),
+				})),
+				expected: {
+					blsKeysUpdate: new Array(100).fill(0).map((_, i) => Buffer.from([i + 3])),
+					bftWeightsUpdate: [BigInt(0), ...new Array(100).fill(0).map((_, i) => BigInt(i + 10))],
+					bftWeightsUpdateBitmap: bytesToBuffer('0'.repeat(27) + '1'.repeat(101)),
+				},
 			},
-		];
-
-		const newValidators: ActiveValidatorWithAddress[] = [
+		],
+		[
+			// only remove validator
```

Comment:
This works, but it will be invalid as there are less than 1 remaining validators.